cmake/modules/lyaml.cmake

```diff
@@ -22,7 +25,6 @@ externalproject_add(
   BUILD_COMMAND ${CMD_MAKE}
   BUILD_IN_SOURCE 1
   BUILD_BYPRODUCTS ${LYAML_LIB}
-  CONFIGURE_COMMAND ./configure --enable-static CFLAGS=-I${LIBYAML_INSTALL_DIR}/include CPPFLAGS=-I${LIBYAML_INSTALL_DIR}/include LDFLAGS=-L${LIBYAML_INSTALL_DIR}/lib LIBS=-lyaml LUA=${LUAJIT_SRC}/luajit LUA_INCLUDE=-I${LUAJIT_INCLUDE}
-  INSTALL_COMMAND sh -c
-  "cp -R ${PROJECT_BINARY_DIR}/lyaml-prefix/src/lyaml/lib/* ${PROJECT_SOURCE_DIR}/userspace/engine/lua"
+  INSTALL_DIR ${LYAML_INSTALL_DIR}
```

Comment:
That is likely the root cause of the issue I have found.

Reply:
Thanks for pointing out the problem. It wasn't specifying INSTALL_DIR though. From reading https://cmake.org/cmake/help/latest/module/ExternalProject.html (skip to INSTALL_DIR) I see "This does not actually configure the external project to install to the given prefix."

The problem was that I need to skip the install step for the project by adding a `INSTALL_CMD ""`. From the same cmake URL, search for INSTALL_CMD and I see "Passing an empty string as the <cmd> makes the install step do nothing."

Looking at the external project definitions, that's what they do, so we should do the same thing here.

I tested after making this change and the lyaml build dir is gone from the packages now.

Reply:
Cool, thank you!